docs/developer-guide/interactivity.md

```diff
@@ -599,7 +599,7 @@ The picking engine returns "picking info" objects which contains a variety of fi
 | `index`  | number | The index of the object in the layer that was picked. |
 | `layer`  | Layer | The top-level layer that the picked object belongs to. Only layers with the `pickable` prop set to true can be picked. |
 | `sourceLayer` | Layer | The immediate layer that rendered the picked pixel. This would be different from `layer` if `layer` is a [CompositeLayer](../api-reference/core/composite-layer.md). |
-| `object` | any | The object that was picked. This is typically an element from the layer's `props.data` array, but can vary from layer to layer. |
+| `object` | any | The object that was picked. This is typically an element from the layer's `props.data` array, but can vary from layer to layer. This field is present only when picking from layers where `props.data` [is an array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#description). |
```

Comment:
Technically this is only true for layers that use the default picking behavior... many composite layers work differently, e.g. GeoJsonLayer's data does not need to be an array, and TextLayer, HexagonLayer etc. implement their own object mapping which does work with Map/Set. I would just say "this field is *usually* only present... unless indicated otherwise in the layer's documentation."